src/libstd/ffi/os_str.rs

```diff
@@ -98,8 +101,36 @@ impl OsString {
 
     /// Extend the string with the given `&OsStr` slice.
     #[stable(feature = "rust1", since = "1.0.0")]
-    pub fn push<T: AsOsStr + ?Sized>(&mut self, s: &T) {
-        self.inner.push_slice(&s.as_os_str().inner)
+    pub fn push<T: As<OsStr>>(&mut self, s: T) {
+        self.inner.push_slice(&s.convert_as().inner)
+    }
+}
+
+#[stable(feature = "rust1", since = "1.0.0")]
+impl From<String> for OsString {
```

Comment:
I wonder if these would be better expressed as `Into`? For example functions may want to take `T: Into<OsString>`.